core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -624,9 +630,13 @@ private void contextProcessedSuccessfully(SearchContext context) {
     }
 
     private void cleanContext(SearchContext context) {
-        assert context == SearchContext.current();
-        context.clearReleasables(Lifetime.PHASE);
-        SearchContext.removeCurrent();
+        try {
+            assert context == SearchContext.current();
+            context.clearReleasables(Lifetime.PHASE);
+            SearchContext.removeCurrent();
+        } finally {
+            context.decRef();
```

Comment:
first we have AssertingIndexSearcher that will report if we have leaks.  I can't assert that it's 0 since it can either be 1 or 0. the default case is 1 since it will only go to 0 once we call freeContext which happens at a later point in time.